### PR TITLE
libreoffice update to 6.4.7.2

### PR DESCRIPTION
**Pull request by dbh201**
update to 6.4.7.2

### CODE REVIEW

File: components/desktop/libreoffice/Makefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ BUILD_BITS=32
 include ../../../make-rules/shared-macros.mk
 
 COMPONENT_NAME=	libreoffice
-COMPONENT_VERSION= 6.4.3
+COMPONENT_VERSION= 6.4.7
 COMPONENT_RC_VERSION= 2
 COMPONENT_REVISION= 4
 COMPONENT_FULL_VERSION=$(COMPONENT_VERSION).$(COMPONENT_RC_VERSION)
```
Comment (AndWac) [inline components/desktop/libreoffice/Makefile:22]: When you increase the COMPONENT_VERSION please reset COMPONENT_REVISION.

**Comment by alarcher** [comment]
@3eka Incremental changes are welcome before jumping to another major, I'd rather see Libreoffice move to 64-bit first.

**Comment by AndWac** [comment]
This doesn't compile here (our older version does):
[C  ] UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.c
test -f /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.c || (echo "Missing generated source file /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.c" && false)
S=/export/home/andreas/oi-userland/components/desktop/libreoffice/libreoffice-6.4.7.2 && B=/export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86 && I=$B/instdir && W=$B/workdir &&  mkdir -p $W/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ $W/Dep/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ && cd /export/home/andreas/oi-userland/components/desktop/libreoffice/libreoffice-6.4.7.2 &&    /usr/gcc/7/bin/gcc -DBOOST_ERROR_CODE_HEADER_ONLY -DBOOST_SYSTEM_NO_DEPRECATED -DCPPU_ENV=gcc3 -DINTEL -DNDEBUG -DOSL_DEBUG_LEVEL=0 -DSUN -DSUN4 -DSYSV -DUNIX -DUNX -D_FILE_OFFSET_BITS=64 -D_POSIX_PTHREAD_SEMANTICS -D_PTHREADS -D_REENTRANT   -fvisibility=hidden    -Wall -Wendif-labels -Wextra -Wstrict-prototypes -Wundef -Wunreachable-code -Wunused-macros -finput-charset=UTF-8 -fmessage-length=0 -fno-common -pipe -fstack-protector-strong  -Wduplicated-cond -Wlogical-op -Wshift-overflow=2 -std=gnu89 -fPIC -Wdeclaration-after-statement -Wshadow   -m32 -O3  -I/usr/include/libjpeg8-turbo -L/usr/lib/libjpeg8-turbo/lib -R/usr/lib/libjpeg8-turbo/lib -fstack-protector -lssp_nonshared  -DHAVE_DLOPEN -D ENABLED_LOCAL_INFILE -D LIBMARIADB -D THREAD -DSQLITE_ENABLE_COLUMN_METADATA=1 -w -DLIBO_INTERNAL_ONLY  -c $W/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.c -o $W/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.o -MMD -MT $W/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.o -MP -MF $W/Dep/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.d_ -I$S/include  -I/usr/local/include  -I$B/config_host  -I$W/UnpackedTarball/mariadb-connector-c/include   && mv $W/Dep/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.d_ $W/Dep/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.d 
{standard input}: Assembler messages:
{standard input}:19: Error: CFI instruction used without previous .cfi_startproc
make[2]: *** [/export/home/andreas/oi-userland/components/desktop/libreoffice/libreoffice-6.4.7.2/solenv/gbuild/LinkTarget.mk:319: /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/GenCObject/UnpackedTarball/mariadb-connector-c/libmariadb/ma_context.o] Error 1
rm /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/line.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/sent.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_prepostdash.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_nodash.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_hu.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word_hu.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/line.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_hu.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_nodash.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/char.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/count_word.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_he.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/count_word.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word_hu.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_prepostdash.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/char.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word_he.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/char_in.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/dict_word_he.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word_he.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/edit_word.txt /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/char_in.brk /export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86/workdir/CustomTarget/i18npool/breakiterator/sent.brk
make[2]: Leaving directory '/export/home/andreas/oi-userland/components/desktop/libreoffice/build/i86'
make[1]: *** [Makefile:282: build] Error 2

**Comment by alarcher** [comment]
@dbh201 I have obsoleted 32-bit gstreamer1 bad and ugly plugins. Do you think it is possible to migrate LO to 64-bit without too much trouble? Just asking in case this is a low-hanging fruit. Thanks.

**Comment by dbh201** [comment]
Ok, I think I can do that. Should I try 6.4.7.2 or 7.2.5.2?

**Comment by alarcher** [comment]
> Ok, I think I can do that. Should I try 6.4.7.2 or 7.2.5.2?

Probably better to try on 6.4.7.2 as you intended to avoid more complications for the time being :)

Also do not hesitate to ping me if you need help :)

**Comment by dbh201** [comment]
It looks like solaris 64bit isn't officially supported. This is from [LibreOffice master: configure.ac](https://github.com/LibreOffice/core/blob/master/configure.ac#L5246):
```
solaris*)
    COM=GCC
    OS=SOLARIS
    RTL_OS=Solaris
    P_SEP=:

    case "$host_cpu" in
    i*86)
        CPUNAME=INTEL
        RTL_ARCH=x86
        PLATFORMID=solaris_x86
        ;;
    sparc)
        CPUNAME=SPARC
        RTL_ARCH=SPARC
        PLATFORMID=solaris_sparc
        ;;
    sparc64)
        CPUNAME=SPARC64
        RTL_ARCH=SPARC64
        PLATFORMID=solaris_sparc64
        ;;
    *)
        AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
        ;;
    esac
    SOLARINC="$SOLARINC -I/usr/local/include"
    ;;
```

**Comment by alarcher** [comment]
> It looks like solaris 64bit isn't officially supported. This is from [LibreOffice master: configure.ac](https://github.com/LibreOffice/core/blob/master/configure.ac#L5246):
> 
> ```
> solaris*)
>     COM=GCC
>     OS=SOLARIS
>     RTL_OS=Solaris
>     P_SEP=:
> 
>     case "$host_cpu" in
>     i*86)
>         CPUNAME=INTEL
>         RTL_ARCH=x86
>         PLATFORMID=solaris_x86
>         ;;
>     sparc)
>         CPUNAME=SPARC
>         RTL_ARCH=SPARC
>         PLATFORMID=solaris_sparc
>         ;;
>     sparc64)
>         CPUNAME=SPARC64
>         RTL_ARCH=SPARC64
>         PLATFORMID=solaris_sparc64
>         ;;
>     *)
>         AC_MSG_ERROR([Unsupported host_cpu $host_cpu for host_os $host_os])
>         ;;
>     esac
>     SOLARINC="$SOLARINC -I/usr/local/include"
>     ;;
> ```

It is a bit misleading probably, the difference is that Solaris x86 was multiarch i386pc and amd64 depending on which compilation flags were passed... unlike Linux and other systems where the arch was split into x86 and x86_64. If you look at https://github.com/LibreOffice/core/blob/master/configure.ac#L9311 the 64-bit case seems to be handled somehow.

For autotools-based software it usually boils down to passing a target either i386-pc-solaris2.11 (32-bit) or  amd64-pc-solaris2.11 (64-bit) otherwise GCC invokes the assembler and linker with the wrong flags.

**Comment by AndWac** [comment]
Can you also update pkg5, please?

**Pull request merged**